src/App.js

```diff
@@ -17,11 +17,23 @@ function App() {
     setContacts([...contacts])
   }
 
+  const sortByPopularity = () => {
+    contacts.sort((a,b) => a.popularity < b.popularity ? 1 : -1);
+    setContacts([...contacts])
+  }
+
+  const sortByName = () => {
+    contacts.sort((a,b) => a.name.localeCompare(b.name));
+    setContacts([...contacts])
+  }
```

Comment:
Another thing to mention. As we cannot edit the state directly your syntax is not exactly right.

Sort() method mutates array, so you want to sort copied array, and only then pass the result into `setContacts()`